addons/point_of_sale/models/pos_config.py

```diff
@@ -206,13 +206,14 @@ def _compute_current_session(self):
         """If there is an open session, store it to current_session_id / current_session_State.
         """
         for pos_config in self:
-            opened_sessions = pos_config.session_ids.filtered(lambda s: not s.state == 'closed')
-            session = pos_config.session_ids.filtered(lambda s: not s.state == 'closed' and not s.rescue)
+            opened_sessions = pos_config.session_ids.filtered(lambda s: s.state != 'closed')
+            rescue_sessions = pos_config.session_ids.filtered(lambda s: s.state != 'closed' and s.rescue)
```

Comment:
you can directly filter on the `opened_sessions` record set above 
```suggestion
            rescue_sessions = opened_sessions.filtered('rescue')
```